sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -53,9 +64,10 @@ def init_gradients_and_hessians(self, n_samples, prediction_dim):
         shape = (prediction_dim, n_samples)
         gradients = np.empty(shape=shape, dtype=G_H_DTYPE)
         if self.hessians_are_constant:
-            # if the hessians are constant, we consider they are equal to 1.
-            # this is correct as long as we adjust the gradients. See e.g. LS
-            # loss
+            # If the hessians are constant, we consider they are equal to 1.
+            # - This is correct for the half LS loss
+            # - For LAD loss, hessians are actually 0, but they are always
```

Comment:
if that's the case, doesn't it make sense to actually set them to `0`? I'm kinda worried about maintainability of this.

Reply:
or even have a parameter in the loss class which is the function which gives you the constant hessians, i.e. for LAD it'd be `np.zeros`.

Reply:
IMO it's more maintainable to have the convention that constant hessians are always 1 rather than have custom constant hessians for each loss. Especially when these hessians are never used, like here.

Reply:
Also, that's something that's always implicit and not explained in the papers, but hessians need to be treated as 1 (even though they're 0): in order for the leaf value computation to be an average instead of a sum.